utils/address.ts

```diff
@@ -13,7 +13,7 @@ export function convertAddress(address: string) {
   return addressMap[address.toLowerCase()] || address.toLowerCase()
 }
 
-export function uniqAddress(_addresses: string[]) {
+export function uniqAddress(_addresses: Array<string | undefined>) {
```

Comment:
It isn't actually changed in this PR, but calling `indexOf` in `flatMap` will perform worst-case O(N²) which may result in degraded performance.

Try something like:

```typescript
export function uniqAddress(addresses: Array<string | undefined>) {
	return [...new Set(addresses.filter(v => typeof v !=== 'undefined'))]
}
```